setupBuild.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+cleanBuild=false
+runBuild=false
+embeddedBuild=false
+debugBuild=false
+runTests=false
+singleJob=false
+while [ $# -ne 0 ]; do
```

Comment:
The previous build script changes I made were bad. This consolidates all build capabilities into a single script. This also allows us to easily add new build flags.